my_enumerable_module.rb

```diff
@@ -1,75 +1,84 @@
 module Enumerable
-    def my_each
-        for i in 0...self.length
-            yield self[i]
-        end
+  def my_each
+    for i in 0...length
+      yield self[i]
     end
-    def my_each_with_index(start_index)
-        for i in 0...self.length
-            yield(self[i],start_index+i)
-        end
-    end
-    def my_select
-        result = []
-        for i in 0...self.length
-            if yield(self[i])
-                result.push(self[i])
-            end
-        end
-        result
-    end
-    def my_all?
-        result = true
-        for i in 0...self.length
-            result = result && yield(self[i])
-        end
-        result
-    end
-    def my_any?
-        result = true
-        for i in 0...self.length
-            result = result || yield(self[i])
-        end
-        result
-    end
-    def my_none?
-        result = false
-        for i in 0...self.length
-            result = result || yield(self[i])
-        end
-        !result
+  end
+
+  def my_each_with_index(start_index = 0)
+    for idx in 0...length
+      yield(self[idx], idx + start_index)
     end
-    def my_count
-        self.length unless block_given?
+    self
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
     end
-end
+    result
+  end
 
-#TESTS
-puts "\nmy_each"
-[1,2,3].each { |x| puts x }
-[1,2,3].my_each {|x| puts x}
+  def my_all?
+    result = true
+    my_each do |item|
+      result &&= yield(item)
+    end
+    result
+  end
 
-puts "\nmy_each_with_index"
-#sets the start index
-[:foo, :bar, :baz].each.with_index(2) { |value, index| puts "#{index}: #{value}" }
-[:foo, :bar, :baz].my_each_with_index(6) { |value, index| puts "#{index}: #{value}" }
+  def my_any?
+    result = true
+    my_each do |item|
+      result ||= yield(item)
+    end
+    result
+  end
 
-puts "\nmy_select"
-puts ["apple","berry","tomatoe","pinnepple","cocoa"].select { |x| x=="apple"}
-puts ["apple","berry","tomatoe","pinnepple","cocoa"].my_select { |x| x=="apple"}
+  def my_none?
+    result = false
+    my_each do |item|
+      result ||= yield(item)
+    end
+    !result
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_none?(String) != [1,2,3,4,5].none?(String)

%w[dog door rod blade].my_none?(/z/) != %w[dog door rod blade].none?(/z/)

%w[dog door rod blade].my_none?(5) != %w[dog door rod blade].none?(5)
```